### PR TITLE
Fixed path checking.

### PR DESCRIPTION
**Pull request by Coder-256**
Fixes cocoapods/cocoapods#5039.

### CODE REVIEW

File: lib/cocoapods-core/source.rb [hunk 1]
```diff
@@ -399,7 +399,7 @@ def to_yaml
     #-------------------------------------------------------------------------#
 
     def ensure_in_repo!
-      return if Pathname.pwd.realpath == repo.realpath
+      return if Pathname.pwd.realpath.to_s.downcase == repo.realpath.to_s.downcase
```
Comment (segiddins) [inline lib/cocoapods-core/source.rb]: This should probably use `File.identical?` in case the user is actually on a case-sensitive FS

**Comment by segiddins** [comment]
Could you also add a changelog entry for this? Thanks!

File: lib/cocoapods-core/source.rb (hunk 1 above)
Comment (Coder-256) [inline lib/cocoapods-core/source.rb]: Done in ab33432.

**Comment by Coder-256** [comment]
@segiddins Done in 9bdc064.

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 ##### Bug Fixes
 
-* None.  
+* Fixed path checking for case-insensitive filesystems.
```
Comment (segiddins) [inline CHANGELOG.md]: trailing two spaces on this line, link to yourself, and then link to the CocoaPods issue (basically just following the format of the other changelog entries)

Comment (Coder-256) [inline CHANGELOG.md]: Ok. I did that in 186d2ac and a92070c. Thank you for helping me so much.

**Pull request merged**